Rely on extension-ci-tools workflow to build linux_amd64_gcc4 extensions

carlopi opened the pull request:
Idea is instead of executing whole workflow job containerized, that means that actions will be run within the container, that in turn means it's hard to  use older images since they will not have the required dependecies versions for the actions (in particular node 20).

Solution is to containerize explicitly the relevant tests in the workflow, invoking actions outside of the container in the regular image that powers the workflow.

Heavy lifting has been done by @samansmink in https://github.com/duckdb/extension-ci-tools/pull/79 and connected PRs.

This PR allows to avoid the need for ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION in Python CI.

carlopi commented:
By examining the before and after runs I see tests are not performed in the in-container step.

This can (and should) be fixed, either here or in a additional PR. For Python workflow in particular this is less of a problem, since tests are performed as part of building and testing on arm64. But this needs to be solved before moving this over to LinuxRelease.yml.

Note for the reviewers that currently code is duplicated, we should consider unifying with extension-ci-tools offering there the relevant callbacks, but I think first having this properly working and the generalizing / simplify.

carlopi commented:
This unblocks the Python side of https://github.com/duckdb/duckdb/pull/14724

samansmink commented on makefiles/build_extensions:
```diff
@@ -0,0 +1,14 @@
+PROJ_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+# Configuration of extension
+EXT_NAME=httpfs
```
EXT_NAME=httpfs?

samansmink commented on makefiles/build_extensions:
```diff
@@ -0,0 +1,14 @@
+PROJ_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
```
I think this makefile is really a bit of a hack, it should not live in a root dir of duckdb/duckdb called makefiles i think. I'm fine with sortof hacking this CI together to get started with moving to using the extension-ci-tools based workflow, but then i'd rather see the makefile live in the `.github` directory to make it clear that its intended for CI use only. Additionally I would like to see a comment in the makefile explaining what its for because its a bit hacky

carlopi replied:
Thanks, had a brief explanation

samansmink reviewed (commented):
Left some comments, really nice that this just works!

carlopi committed:
Rely on extension-ci-tools workflow to build linux_amd64_gcc4 extensions

Idea is instead of executing whole workflow job containerized, that means that actions will
be run within the container, that in turn means it's hard to  use older images since they will
not have the required dependecies versions for the actions (in particular node 20).

Solution is to containerize explicitly the relevant tests in the workflow, invoking actions
outside of the container in the regular image that powers the workflow.

Heavy lifting has been done by @samansmink in https://github.com/duckdb/extension-ci-tools/pull/79
and connected PRs.

This PR allows to avoid the need for ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION in Python CI.

carlopi commented:
Relevant CI run for this is the manylinux_2014 / linux_arm64_gcc4 job that is done as part of Python.yml.

I am not completely sure this propagates correctly the override_git_describe value. I would consider triggering it as if we were preparing for a non-existing release (1.1.4 ?) and double check the compatibility via @hmeriann work. This also would expand the scope of that work to test preprared releases.

Mytherin commented:
Thanks!
